include/ginkgo/core/solver/solver_base.hpp

```diff
@@ -245,6 +401,13 @@ class EnableSolverBase : public SolverBase<MatrixType> {
         this->set_system_matrix_base(new_system_matrix);
     }
 
+    void setup_workspace() const
+    {
+        using traits = solver_workspace_traits<DerivedType>;
+        this->set_workspace_size(traits::num_vectors(*self()),
```

Comment:
This does not allocate yet, right ?

Reply:
no, it just allocates storage for the pointers.